### DISCUSSION TITLE
Is it possible to convert "Shiny for python" to executable file using pyinstaller?????

### DISCUSSION THREAD

**Discussion by HotWesley** [pyinstaller/pyinstaller#8255] [PyInstaller] [answered]
Because of the project, I need to convert Shiny for Python into an exe executable file.

I tried several tests, but I failed. Is this possible? I'm asking you directly because I don't have much data on the Internet.

I'd appreciate it if you could answer me.
Have a nice day :)

**Comment by bwoodsend** [comment]
_Shiny_ here being the [R Shiny Python equivalent](https://shiny.posit.co/py/docs/overview.html)? Given that it's a webserver where the entry point is not your own code, it'll be miserable at best.

**Reply by HotWesley** [reply]
If it's a web server entry point problem, why flask possible? In the video below, he can convert the flask app into an exe file using pyinstaller. 

https://youtu.be/B7zQsw1f-48?si=2QnyhXn4I86xaLcD

Is it possible that the version of Shiny for python issue is still impossible?

**Reply by bwoodsend** [reply]
I never said that makes it impossible – just that it won't be easy – in the same way that freezing flask/django is technically possible but still a terrible choice.

**Reply by HotWesley** [reply]
Shiny mentioned in the question is this below.

https://shiny.posit.co/py/

And the test I did is on the link below.

https://stackoverflow.com/questions/77864783/how-to-convert-shiny-for-python-to-exe-file-using-pyinstaller

It won't be easy, but what can I do if possible? Please give me your opinion.

Plus, what do you think is the best choice for developing desktop apps with Python? Is it PyQt?

**Reply by bwoodsend** [reply, marked as answer]
Well `sys.stderr.write(f"Error: {module_path} not found\n")` is a bug in Shiny. It's assuming that a console is available [which isn't true for windowed applications](https://pyinstaller.org/en/latest/common-issues-and-pitfalls.html#sys-stdin-sys-stdout-and-sys-stderr-in-noconsole-windowed-applications-windows-only). There should be an `if sys.stderr is not None` wrapper around that offending line.

> Plus, what do you think is the best choice for developing desktop apps with Python? Is it PyQt?

I use PyQt. I don't particularly like it. It makes more sense than tkinter. It's more Windows-friendly than GTK. I've not really tried any others.

**Reply by HotWesley** [reply]
I can't fix it because it's a bug inside shiny package... Thank you for your answer :)
